e2e/vuln_1783967524.py

```
@@ -0,0 +1,14 @@
+import hashlib, sqlite3
+
+def make_token(password):
+    # Weak hashing — MD5 is not suitable for secrets
+    return hashlib.md5(password.encode()).hexdigest()
```

Ruff (0.15.21)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)